RealmSwift/SwiftUI.swift

```diff
@@ -237,6 +242,7 @@ private class ObservableStorage<ObservedType>: ObservableObject where ObservedTy
         willSet {
             if newValue != value {
                 objectWillChange.send()
+                objectWillChange.update(value: newValue)
```

Comment:
When `ObservedResults` is setup the publisher is passed `RLMResults<ResultType>.emptyDetached()`. When the actual `Results` to be observed is ready we need to update the publisher to hold a reference to it. Otherwise `receive()` will never setup the `ObservationSubscription` again once the original token is invalidated.